o1vm/src/interpreters/riscv32im/interpreter.rs

```diff
@@ -1,3 +1,26 @@
+//! This module implement an interpreter for the RISCV32 IM instruction set
+//! architecture.
+//!
+//! The implementation mostly follows (and copy) code from the MIPS interpreter
+//! available [here](../mips/interpreter.rs).
+//!
+//! ## Credits
+//!
+//! We would like to thank the authors of the following documentations:
+//! - <https://msyksphinz-self.github.io/riscv-isadoc/html/rvm.html> ([CC BY
+//! 4.0](https://creativecommons.org/licenses/by/4.0/)) from
```

Comment:
Reviewer: check it fits the license requirement.

Reply:
it fits the license requirement according to https://creativecommons.org/licenses/by/4.0/